tests/filecheck/mlir-conversion/builtin_attrs.mlir

```diff
@@ -0,0 +1,79 @@
+// RUN: xdsl-opt %s -t mlir | xdsl-opt -f mlir -t mlir | FileCheck %s
+
+"builtin.module"() ({
+  "func.func"() ({
+    ^bb0(%arg0: index):
+    "func.return"() : () -> ()
+  }) {function_type = (index) -> (), sym_name = "index_type"} : () -> ()
+
+  // CHECK: (index)
+
+
+  "func.func"() ({
+    ^bb0(%arg0: i32, %arg1: i64, %arg2: i1):
+    "func.return"() : () -> ()
+  }) {function_type = (i32, i64, i1) -> (), sym_name = "int_type"} : () -> ()
+
+  // CHECK: (i32, i64, i1)
+
+
+  "func.func"() ({
+    ^bb0(%arg0: f16, %arg1: f32, %arg2: f64):
+    "func.return"() : () -> ()
+  }) {function_type = (f16, f32, f64) -> (), sym_name = "float_type"} : () -> ()
+
+  // CHECK: (f16, f32, f64)
+
+
+  "func.func"() ({}) {function_type = () -> (), value = 42.0 : f32, sym_name = "float_attr"} : () -> ()
```

Comment:
I didn't know these kinds of functions are possible in MLIR. The more you know!

Reply:
What do you mean by these kind of functions?
I just added the `value` attribute to be able to use any attribute I wanted. value has no semantics here ^^'.
But I'll actually try to see if MLIR can parse this file, I haven't tried it!

Reply:
Ah i see. I thought this were some kind of functions that would always return the same thing :sweat_smile: